test/e2e/operator_test.go

```diff
@@ -70,172 +76,175 @@ var _ = Describe("Operator API", func() {
 	// 14. Ensure the reference to ns-a is eventually removed from o's status.components.refs field
 	// 15. Delete o
 	// 16. Ensure o is not re-created
-	It("should surface components in its status", func() {
-		o := &operatorsv1.Operator{}
-		o.SetName(genName("o-"))
-
-		Consistently(o).ShouldNot(ContainCopiedCSVReferences())
-
-		Eventually(func() error {
-			return client.Create(clientCtx, o)
-		}).Should(Succeed())
+	Context("when an Operator resource can select its components by label", func() {
+		var (
+			o *operatorsv1.Operator
+		)
+		BeforeEach(func() {
+			o = &operatorsv1.Operator{}
+			o.SetName(genName("o-"))
 
-		defer func() {
 			Eventually(func() error {
-				err := client.Delete(clientCtx, o)
-				if apierrors.IsNotFound(err) {
-					return nil
-				}
-
-				return err
+				return client.Create(clientCtx, o)
+			}).Should(Succeed())
+		})
+		AfterEach(func() {
+			Eventually(func() error {
+				return controllerclient.IgnoreNotFound(client.Delete(clientCtx, o))
 			}).Should(Succeed())
-		}()
+		})
 
-		By("eventually having a status that contains its component label selector")
-		w, err := operatorClient.Watch(clientCtx, listOpts)
-		Expect(err).ToNot(HaveOccurred())
-		defer w.Stop()
+		It("should not contain copied csv status references", func() {
+			Consistently(o).ShouldNot(ContainCopiedCSVReferences())
```

Comment:
I moved this check to it's own It spec as before we were checking whether the operator object had any CSV references _before_ creating that resource, leading to what appeared to be a wasted check.